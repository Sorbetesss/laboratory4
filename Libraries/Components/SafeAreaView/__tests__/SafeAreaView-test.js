/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow
 */

'use strict';

const React = require('React');
const SafeAreaView = require('SafeAreaView');
const render = require('../../../../jest/renderer');

describe('<SafeAreaView />', () => {
  it('should set displayName to prevent <Component /> regressions', () => {
    expect(SafeAreaView.displayName).toBe('SafeAreaView');
  });

  it('should render as <SafeAreaView> when mocked', () => {
    const instance = render.create(<SafeAreaView />);
    expect(instance).toMatchSnapshot();
  });

  it('should shallow render as <ForwardRef(SafeAreaView)> when mocked', () => {
    const output = render.shallow(<SafeAreaView />);
    expect(output).toMatchSnapshot();
  });

  it('should shallow render as <ForwardRef(SafeAreaView)> when not mocked', () => {
    jest.dontMock('SafeAreaView');

    const output = render.shallow(<SafeAreaView />);
    expect(output).toMatchSnapshot();
  });

  it('should render as <SafeAreaView> when not mocked', () => {
    jest.dontMock('SafeAreaView');

    const instance = render.create(<SafeAreaView />);
    expect(instance).toMatchSnapshot();
  });
});
