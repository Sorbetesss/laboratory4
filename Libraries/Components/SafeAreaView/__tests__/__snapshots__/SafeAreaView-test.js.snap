// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SafeAreaView /> should render as <SafeAreaView> when mocked 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
/>
`;

exports[`<SafeAreaView /> should render as <SafeAreaView> when not mocked 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
/>
`;

exports[`<SafeAreaView /> should shallow render as <ForwardRef(SafeAreaView)> when mocked 1`] = `<ForwardRef(_SafeAreaView) />`;

exports[`<SafeAreaView /> should shallow render as <ForwardRef(SafeAreaView)> when not mocked 1`] = `<ForwardRef(_SafeAreaView) />`;
